src/qibolab/_core/native.py

```diff
@@ -30,18 +30,35 @@ def _normalize_angles(theta, phi):
 
 
 def rotation(
-    seq: PulseSequence, theta: float = np.pi, phi: float = 0.0
+    seq: PulseSequence,
+    theta: float = np.pi,
+    phi: float = 0.0,
+    rx90: bool = False,
 ) -> PulseSequence:
     """Create a sequence for single-qubit rotation.
 
     ``theta`` will be the angle of the rotation, while ``phi`` the angle that the rotation axis forms with x axis.
+    If ``rx90`` is True the rotation will be performed starting from an RX90 pulse doubling its amplitude,
+    if the amplitude is greater than 0.5 two pulses are played.
     """
     theta, phi = _normalize_angles(theta, phi)
     ch, pulse = seq[0]
     assert isinstance(pulse, Pulse)
-    new_amplitude = pulse.amplitude * theta / np.pi
+    amplitude = pulse.amplitude * theta / np.pi
+    if rx90:
+        if amplitude > 0.5:
+            return PulseSequence(
+                [
+                    (ch, replace(pulse, amplitude=amplitude, relative_phase=phi)),
+                    (ch, replace(pulse, amplitude=amplitude, relative_phase=phi)),
+                ]
+            )
```

Comment:
```suggestion
            p = replace(pulse, amplitude=amplitude, relative_phase=phi)
            return PulseSequence([(ch, p), (ch, p.model_copy())])
```

Probably, we could even use the exact same pulse, just doing `[(ch,p)]*2`. But if you do not have independent objects, you will have to sweep them simultaneously (which is not even such a bad thing, since in principle they should be acting as one...).

Reply:
I see your point, for the time being let's keep them separated.